### PR TITLE
[uni-algo] Add new port

### LINKED ISSUE #29716: [New Port Request] uni-algo
Library name: uni-algo

Library description: Unicode Algorithms Implementation

Source repository URL: https://github.com/uni-algo/uni-algo

### PR DESCRIPTION
**Pull request by mg152**
Fixes #29716

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

Tested on:
Windows MSVC 19.16 (triplets: `x86-windows`, `x64-windows`, `x64-windows-release`)
Windows MinGW GCC 12.1 (triplets: `x64-mingw-static`, `x64-mingw-dynamic`)
Linux GCC 12.2 (triplets: `x64-linux`, `x64-linux-dynamic`, `x64-linux-release`)
Tested in both classic and manifest modes of vcpkg.
Everything seems to be fine.

### CODE REVIEW

**Comment by mg152** [comment]
@microsoft-github-policy-service agree

File: ports/uni-algo/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,27 @@
+vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
```
Comment (FrankXie05) [inline ports/uni-algo/portfile.cmake:1]:
```suggestion
```
It's not necessary now.

File: ports/uni-algo/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    HEAD_REF main
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(NO_PREFIX_CORRECTION)
```
Comment (FrankXie05) [inline ports/uni-algo/portfile.cmake:17]:
```suggestion
vcpkg_cmake_config_fixup(NO_PREFIX_CORRECTION)

```
`NO_PREFIX_CORRECTION` Currently the correction does not take into account how the files are moved, and applies a rather simply correction which in some cases will yield the wrong results.

File: ports/uni-algo/usage [hunk 3]
```diff
@@ -0,0 +1,4 @@
+The package uni-algo provides CMake targets:
```
Comment (FrankXie05) [inline ports/uni-algo/usage:1]: I neet to test the usage.

File: ports/uni-algo/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "uni-algo",
+  "version-semver": "0.7.0",
+  "description": "Unicode Algorithms Implementation",
+  "homepage": "https://github.com/uni-algo/uni-algo",
+  "license": "Unlicense OR MIT",
```
Comment (FrankXie05) [inline ports/uni-algo/vcpkg.json:6]:
```suggestion
  "license": "Unlicense OR MIT",
```
If there is no license, the value is `null`.

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/uni-algo/portfile.cmake (hunk 2 above)
Comment (mg152) [reply ports/uni-algo/portfile.cmake:15]: `vcpkg_cmake_config_fixup` breaks paths in CMake install targets if used without `NO_PREFIX_CORRECTION`.

File: ports/uni-algo/vcpkg.json (hunk 4 above)
Comment (mg152) [reply ports/uni-algo/vcpkg.json:6]: I'm not sure what you meant. The license matches what upstream says.

Comment (FrankXie05) [reply ports/uni-algo/vcpkg.json:6]: Upstream `Public Domain License` means to declare the license of the current library, and to be under the license of `MIT`.
The license should be `MIT`.

File: ports/uni-algo/portfile.cmake (hunk 2 above)
Comment (FrankXie05) [reply ports/uni-algo/portfile.cmake:15]: If it breaks, it means that the upstream cmake config file has not been downloaded to the default or specified correct location.

File: ports/uni-algo/usage (hunk 3 above)
Comment (FrankXie05) [reply ports/uni-algo/usage:1]: The usage has been tested successfully.

File: ports/uni-algo/vcpkg.json (hunk 4 above)
Comment (dg0yt) [reply ports/uni-algo/vcpkg.json:6]: AFAICS the Public Domain License is correctly identified as `Unlicense`.  
https://tools.spdx.org/app/check_license/ 
So `"Unlicense OR MIT"` should be correct.

File: ports/uni-algo/portfile.cmake (hunk 2 above)
Comment (mg152) [reply ports/uni-algo/portfile.cmake:17]: If I understand correctly how vcpkg works it will be ON anyway, but I added it just in case.

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**